src/ui/viewer/CommandViewer.tsx

```
@@ -0,0 +1,292 @@
+
+    // Keep tree in sync with commands from store
+    useEffect(() => {
+        setItems(commandsToTreeItems(commands));
+    }, [commands]);
+
+    // Create flattened version of tree for dndkit
+    const flattenedItems = useMemo(() => {
+        const flattenedTree = flattenTree(items);
+        const collapsedItems = flattenedTree.reduce<string[]>(
+            (acc, { children, collapsed, id }) =>
+                collapsed && children.length ? [...acc, id as string] : acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)